doc/release/upcoming_changes/22539.deprecation.rst

```diff
@@ -0,0 +1,24 @@
+``np.find_common_type`` is deprecated
```

Comment:
This is a lot of info for a release note but then again the situation is complicated. I'm +1 for leaving the info in the release notes instead of the docs proper since this is a deprecation and the complexity will be reduced in the future.